packages/react-components/src/components/GridLayout.tsx

```diff
@@ -31,43 +24,99 @@ export interface GridLayoutProps {
   styles?: BaseCustomStylesProps;
 }
 
-const calculateStandardLayoutRows = (numberOfItems: number, gridCol: number): number =>
-  Math.ceil(numberOfItems / gridCol);
-
-const calculateStandardLayoutColumns = (numberOfItems: number): number =>
-  numberOfItems > 0 ? Math.ceil(Math.sqrt(numberOfItems)) : 1;
-
 /**
- * A component to lay out audio / video tiles in a call.
+ * A component to lay out audio / video participants tiles in a call.
  *
  * @public
  */
 export const GridLayout = (props: GridLayoutProps): JSX.Element => {
-  const [gridCol, setGridCol] = useState(1);
-  const [gridRow, setGridRow] = useState(1);
-
-  const { children, layout = 'standard', styles } = props;
+  const { children, styles } = props;
   const numberOfChildren = React.Children.count(children);
 
-  switch (layout) {
-    case 'standard': {
-      const numberOfColumns = calculateStandardLayoutColumns(numberOfChildren);
-      if (gridCol !== numberOfColumns) setGridCol(numberOfColumns);
-      const numberOfRows = calculateStandardLayoutRows(numberOfChildren, gridCol);
-      if (gridRow !== numberOfRows) setGridRow(numberOfRows);
-      break;
+  const targetRef = useRef<HTMLDivElement>(null);
+  const [blockProps, setBlockProps] = useState<BlockProps>({
+    horizontal: true,
+    numBlocks: Math.ceil(Math.sqrt(numberOfChildren))
+  });
+
+  useEffect(() => {
+    const updateBlockProps = (): void => {
+      if (targetRef.current) {
+        setBlockProps(
+          calculateBlockProps(numberOfChildren, targetRef.current.offsetWidth, targetRef.current.offsetHeight)
+        );
+      }
+    };
+    const observer = new ResizeObserver(updateBlockProps);
+    if (targetRef.current) {
+      observer.observe(targetRef.current);
     }
+    updateBlockProps();
+    return () => observer.disconnect();
+  }, [numberOfChildren, targetRef.current?.offsetWidth, targetRef.current?.offsetHeight]);
+
+  return (
+    <div ref={targetRef} className={mergeStyles(gridLayoutContainerStyle, styles?.root)}>
+      {createBlocks({ ...blockProps, children })}
+    </div>
+  );
+};
+
+/**
+ * Props to create blocks for children in Grid Layout
+ *
+ */
+export type BlockProps = {
+  horizontal: boolean;
+  numBlocks: number;
+};
+
+const createBlocks = (props: BlockProps & { children: React.ReactNode }): React.ReactNode => {
```

Comment:
This is a standard react component. Let's name it and use it as such
Example, `Blocks`

Reply:
Removed this because `createBlocks` was causing so much unnecessary re-rendering.